src/osd/PG.h

```diff
@@ -186,6 +186,10 @@ class PG : public DoutPrefixProvider, public PeeringState::PeeringListener {
   /// scrubbing state for both Primary & replicas
   bool is_scrub_active() const { return m_scrubber->is_scrub_active(); }
 
+  bool get_reserve_failed() const { return m_scrubber->get_reserve_failed(); }
```

Comment:
I would prefer, if we choose to have this status flag, to have it as part of the object that actually holds that status -
m_local_osd_resource.

Reply:
What do you mean by object?  This is really the PG that is marked.

Reply:
The status of "being locked for scrubbing" is now managed - indeed inside the PG, but within sub-elements of the
PgScrubber. It shouldn't be spread out.

Reply:
What does "spread out" mean?

Reply:
What I meant is this: shouldn't the status of 'I have failed trying to lock my replicas' be managed by the same
object that manages the status 'what replicas I have tried to have resource-locked for me, and what's the status with
that locking?' (ReplicaReservations)

(I was wrong when ref to m_local_osd_resource (see my question on that below))